kong/plugins/hmac-auth/access.lua

```diff
@@ -177,6 +198,16 @@ local function do_authentication(conf)
     return false, {status = 403, message = SIGNATURE_NOT_VALID}
   end
 
+  -- If request body validation is enabled, then verify hash of request body.
+  if conf.validate_request_body then
```

Comment:
Also we should validate headers first and then body. As if header is tampered there is no point validating body.